static/js/message_flags.js

```diff
@@ -113,6 +113,19 @@ exports.toggle_starred_and_update_server = function (message) {
     }
 };
 
+exports.unstar_all_messages = function () {
+    var starred_msg_ids = page_params.starred_messages;
+    channel.post({
+        url: '/json/messages/flags',
+        idempotent: true,
+        data: {
+            messages: JSON.stringify(starred_msg_ids),
+            flag: 'starred',
+            op: 'remove',
```

Comment:
@timabbott FYI, done in bulk. This should have came to my mind when I saw [message.id] being passed in request. Certainly we can add more items to list. It just happened. :(